module/zfs/dmu_tx.c

```diff
@@ -1089,10 +1103,11 @@ dmu_tx_wait(dmu_tx_t *tx)
 			DMU_TX_STAT_BUMP(dmu_tx_dirty_over_max);
 		while (dp->dp_dirty_total >= zfs_dirty_data_max)
 			cv_wait(&dp->dp_spaceavail_cv, &dp->dp_lock);
+		last_smooth = dp->dp_last_smooth;
```

Comment:
Why is the read of `dp_last_smooth` not being done in `dmu_tx_delay`, like the update?

Is it because we need to hold the mutex, and don't want to pay the cost of re-acquiring in `dmu_tx_delay` just to read the value?
Does it actually make a perf difference in practice?

Reply:
Yes, its because we would need to take again a mutex.
This method is equivalent to dp_dirty_total which also requires mutex.